scheduler/pkg/kafka/dataflow/server_test.go

```diff
@@ -75,3 +77,46 @@ func TestCreateTopicSources(t *testing.T) {
 		})
 	}
 }
+
+func TestCreatePipelineTopicSources(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	type test struct {
+		name         string
+		server       *ChainerServer
+		pipelineName string
+		inputs       []string
+		sources      []*chainer.PipelineTopic
+	}
+
+	tests := []test{
+		{
+			name: "misc inputs",
+			server: &ChainerServer{
+				logger:     log.New(),
+				topicNamer: kafka.NewTopicNamer("default"),
+			},
+			pipelineName: "p1",
+			inputs: []string{
+				"foo.inputs",
+				"foo.outputs",
+				"foo.step.bar.inputs",
+				"foo.step.bar.outputs",
+				"foo.step.bar.inputs.tensora",
+			},
+			sources: []*chainer.PipelineTopic{
+				{PipelineName: "foo", TopicName: "seldon.default.pipeline.foo.inputs"},
+				{PipelineName: "foo", TopicName: "seldon.default.pipeline.foo.outputs"},
+				{PipelineName: "foo", TopicName: "seldon.default.model.bar.inputs"},
+				{PipelineName: "foo", TopicName: "seldon.default.model.bar.outputs"},
+				{PipelineName: "foo", TopicName: "seldon.default.model.bar.inputs.tensora"},
```

Comment:
question (related to another comment I added): why to we want to use input topics in downstream pipelines? it seems we should only care about outputs?

Reply:
I was assuming any step of a pipeline can be reference and not to restrict people. There is a bigger issue for the TensorMap ability to change tensor names as allowing any inputs mean one can choose the same model from two different Pipelines to join to but the tensor map can't distinguish this as its just impemented as tensor names from a model, not a model in a pipeline.

Reply:
All the examples we have so far user output topics and logically it makes sense to use output topics.

Would there be any realistic use case for consuming wiring input topic -> external pipeline input? 
Maybe a shadow pipeline that gets copy from the main pipeline to do something difference.
Ok I know realised that perhaps in my mind there might not be clear in my mind, there could be in the future
a use case that I havent thought about.

Reply:
You could be doing drift analysis on model inputs as one example. I don't think it makes sense to restrict at present.